lib/src/entity.dart

```diff
@@ -60,4 +62,34 @@ abstract class Entity extends PositionComponent {
   bool hasBehavior<T extends Behavior>() {
     return findBehavior<T>() != null;
   }
+
+  /// Adds the given [behavior] to this entity only if the entity does not yet
+  /// have the behavior.
+  Future<void>? addBehavior<T extends Behavior>(T behavior) {
+    assert(
+      !hasBehavior<T>(),
```

Comment:
Technically this could still be out of async if users don't await these calls.

One solution would be to make the method truly async. Anyone has other suggestions?

Reply:
Yeah why not just return Future<void>?

Reply:
Because I kept it in line with Flame API, which is also `Future<void>?`, but I would love it if we can choose to not do that. As that would ensure the rule much better.